src/python/ensembl/io/genomio/genome_metadata/dump.py

```diff
@@ -151,18 +154,27 @@ def filter_genome_meta(
     return filtered_metadata
 
 
-def check_assembly_refseq(gmeta_out: Dict[str, Any]) -> None:
+def check_assembly_refseq(gmeta_out: dict[str, Any]) -> None:
     """Update the GCA accession to use GCF if it is from RefSeq.
 
     Args:
         genome_metadata: Nested metadata key values from the core metadata table.
     """
     assembly = gmeta_out.get("assembly", {})
-    if assembly.get("provider_name", "") == "RefSeq":
-        assembly["accession"] = assembly["accession"].replace("GCA", "GCF")
+    if bool(assembly.get("provider_name")):
+        if assembly.get("provider_name", "") == "RefSeq":
```

Comment:
At this point you are certain `provider_name` is defined and has a non-empty value:
```suggestion
        if assembly["provider_name"] == "RefSeq":
```